luigi/contrib/hive.py

```diff
@@ -180,8 +180,12 @@ class MetastoreClient(HiveClient):
     def table_location(self, table, database='default', partition=None):
         with HiveThriftContext() as client:
             if partition is not None:
-                partition_str = self.partition_spec(partition)
-                thrift_table = client.get_partition_by_name(database, table, partition_str)
+                try:
+                    import hive_metastore.ttypes
+                    partition_str = self.partition_spec(partition)
+                    thrift_table = client.get_partition_by_name(database, table, partition_str)
+                except hive_metastore.ttypes.NoSuchObjectException as e:
```

Comment:
I think you should just leave it as it is. Any dependency in luigi.contrib shouldn't be in requirements.txt. Optionally throw a warning if the right package isn't installed (and you should probably add a note that installing hive-thrift-py is highly recommended)